PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/LoadJsonConfigFile.java

```diff
@@ -21,43 +30,158 @@ public class LoadJsonConfigFile {
-        return jsonObject;
+        return JSON_OBJECT;
+    }
+
+
+    /**
+     * Base Method to get the value for a given rule name and key as an Object
+     * If the rule name exists and has the key, return the value
+     * eg getConfigValue("ConnectionStringCheck", "method_to_check")
+     * this will return the value for the key "method_to_check" in the rule "ConnectionStringCheck"
+     *
+     * @param ruleName String rule name
+     * @param key String key
+     * @return Object value for the key
+     */
+    private Map<String, String> getRuleConfigurationValue(String ruleName, String key) {
+        Map<String, String> resultMap = new HashMap<>();
+        if (JSON_OBJECT.has(ruleName)) {
+            JSONObject ruleConfig = JSON_OBJECT.getJSONObject(ruleName);
+            if (ruleConfig.has(key)) {
+                // Assuming all values are Strings or can be represented as Strings
+                resultMap.put(key, ruleConfig.getString(key));
+            }
+        }
+        return resultMap;
+    }
+
+    /**
+     * Method to get the methods to check for a given rule name, which is a List of Strings
+     * or a singular String
+     * @param ruleName String rule name
+     * @return List<String> methods to check. If the value is a singular String, return a List with one element
+     */
+    public List<String> getMethodsToCheck(String ruleName) {
+        String key = "methods_to_check";
+
+        if (JSON_OBJECT.has(ruleName)) {
+            JSONObject ruleConfig = JSON_OBJECT.getJSONObject(ruleName);
+            if (ruleConfig.has(key) && ruleConfig.get(key) instanceof String) {
+                // If it's a string, convert it to a list with one element
+                String method = ruleConfig.getString(key);
+                return List.of(method);
+            } else if (ruleConfig.has(key) && ruleConfig.get(key) instanceof JSONArray) {
+                // If it's a JSONArray, process it
+                return processMethodsToCheck(ruleConfig.getJSONArray(key));
+            } else {
+                LOGGER.log(Level.WARNING, "No methods to check found for rule: " + ruleName);
+            }
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Method to get the client name for a given rule name
+     * @param ruleName String rule name
+     * @return String client name
+     */
+    public String getClientName(String ruleName) {
+        String key = "client_name";
+        Map<String, String> ruleConfig = getRuleConfigurationValue(ruleName, "client_name");
+        return ruleConfig.get(key);
+    }
+
+    /**
+     * Method to get the suggestion message for a given rule name
+     * @param ruleName String rule name
+     * @return String suggestion message
+     */
+    public String getAntiPatternMessage(String ruleName) {
+        String key = "antipattern_message";
+        Map<String, String> ruleConfig = getRuleConfigurationValue(ruleName, "antipattern_message");
```

Comment:
Should we not store the result and then just use a getter for that map key to retrieve the corresponding result?